app/src/main/java/org/mozilla/reference/browser/BrowserApplication.kt

```diff
@@ -8,11 +8,21 @@ import android.app.Application
 import mozilla.components.support.base.log.Log
 import mozilla.components.support.base.log.sink.AndroidLogSink
 import org.mozilla.reference.browser.ext.isCrashReportActive
+import android.app.ActivityManager
+import android.content.Context
 
 class BrowserApplication : Application() {
     val components by lazy { Components(this) }
 
     override fun onCreate() {
+        // We only want to run this for the application process and not for sub
+        // processes e.g. the Gecko :tab process. Otherwise we'd be initializing
+        // our components multiple times and we'd also create the GeckoEngine
+        // and GeckoRuntime in the wrong process.
+        if (isChildProcess()) {
+            return
+        }
+
         super.onCreate()
```

Comment:
you still want to call `super.onCreate` I think

Reply:
Yeah, good catch. Updated.